src/features_creator/features_creator.py

```diff
@@ -38,3 +38,54 @@ def get_matching_column_names(data, pattern):
 
     """
     ...
+
```

Comment:
Could you unstage the changes you made to this file so that we only merge the CSV with the data? Once that's done, I can approve

Reply:
Closing this PR, I will open a new PR with all the changes from the main branch.